Update examples and structure

tedenda opened the pull request:
I have started to update the example folder and it's structure, moved pictures to image folder, updated README.md files etc.
Also started to make a README.md for OneWire sensors.

Please give some feedback on this @mairas @ba58smith, is this the way to go with all examples?

mairas commented:
I'd say it looks really good! Are you planning to eventually have all examples in their own directories or keep the mixed approach as now?

ba58smith commented:
I'd leave the single-file examples all in the /examples directory, and make sub-directories only for those examples with multiple files. Most examples won't have multiple files, so this approach will make it quicker to get to most example files. (But only by one click... so I'm not terribly opposed to having each in its own directory.)

tedenda commented:
> I'd say it looks really good! Are you planning to eventually have all examples in their own directories or keep the mixed approach as now?

> I'd leave the single-file examples all in the /examples directory, and make sub-directories only for those examples with multiple files. Most examples won't have multiple files, so this approach will make it quicker to get to most example files. (But only by one click... so I'm not terribly opposed to having each in its own directory.)

I have not thought about this so much.
But for now I will only move files as I make a guide.

tedenda commented on examples/onewire_temoerature/README.md:
```diff
@@ -0,0 +1,59 @@
+# Onewire temperature sensor(s)
```
> Can't comment on the filename: you have a typo in the directory name! :-D

Thanks!
Will fix it!

tedenda replied:
Done!

tedenda commented:
So, a lot of writing, capturing pictures, edit code, moving and renaming files etc.
I am happy with it so far, I hope this will help new people to success with there first OneWire node!

Please review it and give your comments, @ba58smith @mairas.
I am sure there is misspelled words and other things that should be fixed before merge.

I think it's best to make this PR golden and merge it.
Then I start with a new sensor, keep the PR tighter than this, this is all over the place, sorry!

ba58smith commented on examples/onewire_temoerature/OneWire_temperatures.cpp:
```diff
@@ -17,20 +17,26 @@ ReactESP app([]() {
   DallasTemperatureSensors* dts = new DallasTemperatureSensors(D7);
 
+  // Define how often SensESP should read the sensor(s)
   uint read_delay = 500;
```
You might want to add ", in milliseconds" to the end of the above comment.

tedenda replied:
I think it's good to make examples informative, it's now added, thanks!

ba58smith commented on examples/onewire_temoerature/OneWire_temperatures.cpp:
```diff
@@ -17,20 +17,26 @@ ReactESP app([]() {
   uint read_delay = 500;
 
+  // Measure colant temperature
   auto* coolant_temp =
```
"coolant". (Two o's.)

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temoerature/README.md:
```diff
@@ -0,0 +1,28 @@
+ ## Introduction
+
+This example implements onewire temperature sensors usig 'Dallas DS18B20 temperature sensor' connected to a Wemos D1-Mini running SensESP to communicate the engine cooling temperature wirelessly to the SignalK navigation system.
+
```
"using" (not "usig").

Is Signal K (with a space, by the way) a "navigation system"?

tedenda replied:
Corrected
Changed to "Signal K server"

ba58smith commented on examples/fuel_level_sensor/README.md:
```diff
@@ -1,11 +1,15 @@
-The circuit required to measure the resistance is based on this: [voltage devider calculator](http://www.ohmslawcalculator.com/voltage-divider-calculator); basically you connect 1 transducer wire to ground (GND), another one to A0 (SIGNAL) and put a resistor between the 3.3v rail and the SIGNAL wire (in series with A0). The circuit works because the resistor is a known value, hence we can deduce  ...
+## Connection
+The circuit required to measure the resistance is based on this: [voltage devider calculator](http://www.ohmslawcalculator.com/voltage-divider-calculator). Basically you connect 1 transducer wire to ground (GND), another one to A0 (SIGNAL) and put a resistor between the 3.3v rail and the SIGNAL wire (in series with A0). The circuit works because the resistor is a known value, hence we can deduce  ...
 
```
Do you need to say the value of the resistor, or is that done somewhere else?

tedenda replied:
Is this a better formulation:
"The circuit works because the resistence of the fixed resistor is a known value, hence we can deduce the value of the transducer."

I think i'll add a picture form frizzing to make the circuit more clear, is't a good idea? 

Also found "devider" corrected to "divider".

ba58smith replied:
That works. Just make sure to spell it "resistance" rather than "resistence".

tedenda replied:
I am sorry for my bad spelling...
Hopefully I learn a bit doing this 😀. Thanks for your help!

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+## Hardware
+
+The raw DS18B20 sensor from Maximin looks like this:
+
```
Maxim, or Maxim Integrated. Not "Maximin".

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+![Circuit](images/circuit.png "Circuit")
+
+You can connect 1 to "many" sensors to one SensESP, but be carefull with long thinn cables, that can eventually cause problems.
+
```
"thin", not "thinn".

I've not had trouble with any DS18B20 cables as long at 3 meters. If you know of some cable length limitations, you should be more specific than "long thin cables". What does that mean, exactly?

tedenda replied:
I have worked with another firmware for ESP's, ESP-Easy.
There was a test done, see under "Precautions" on this page: [ESP-Easy Wiki](https://www.letscontrolit.com/wiki/index.php?title=Dallas_DS18b20)

What to do?
- Copy that text and put a reference to the source
- Only reference to the Wiki
- Don't mention it at all (remove "but be carefull with long thinn cables, that can eventually cause problems")

ba58smith replied:
I would say something like "Wire lengths greater than X meters can cause problems. See https://the_url_to_the_ESP_Easy_wiki.com."

tedenda replied:
Thanks!

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+
+### Step 3
+SenseESP need to know where the sensor(s) are connected to the board, edit this section in the code:
+
```
"needs".

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+If you for example only have 2 sensors you remove the two last sensors in the code, if you have more, copy and paste in more sensors.
+
+Rename sensors to what the they measure according to the selected places in this picture:
+
```
Extra "the" in the above sentence.

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+Rename sensors to what the they measure according to the selected places in this picture:
+
+![EditNamneHere](images/edit_name_here.png "edit namne here")
+
```
"name", not "namne".

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+
+### Step 5
+It's time to upload the code to the ESP, make sure it's connected to the computer. Click "Upload and Monitor All", firmware will now be built and uploaded to the board if no errors is found. Onece the firmware is uploaded the monitor will start showing what the SenseESP is doing.
+
```
"are found", not "is found".

"Once", not "Onece.

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
````diff
@@ -0,0 +1,235 @@
+␛[0m(I) (load_configuration) Not loading configuration: file does not exist: /12vAltTemperature/linear
+```
+Here is all paths that will be reported to SignalK server:
+```
````
"Here are", not "Here is".

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+
+### Step 7
+Now the SenseESP is started and are waiting to be configured. Connect to the "Configure SensESP" network, short after the webbrowser should start and you see this:
+
```
"Is waiting".

Connect to the "Configure SensESP" network. Shortly after you connect, a web page should open and you should see the image below. If a web page doesn't open, you may have to manually open a page to 192.168.4.1.

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+![CaptivePortal2](images/captiveportal2.png "captive portal 2")
+
+Click on your network and some info will be filled in for you, fill in missing info and give the unit a name:
+
```
Replace "unit" with "MCU" or "ESP Device".

"... a name, like "EngineTemp" or some other meaningful name.

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+![CaptivePortal3](images/captiveportal3.png "captive portal 3")
+
+Click save and the SensESP will reboot and try to connect to your WiFi. If something is wrong it will appear as an aacesspoint again and you can correct settings.
+If all is good the SensESP will connect to your WiFi and search for your SignalK server.
```
"MCU" or "ESP Device" - not "SensESP".

"access point", not "aacesspoint".

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
````diff
@@ -0,0 +1,235 @@
+```
+
+Here you can see that SenseESP is connecting to wifi openplotter and get IP 10.10.10.153:
+```
````
"gets".

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
````diff
@@ -0,0 +1,235 @@
+```
+
+Here is the server found at ip 10.10.10.1 port 3000:
+```
````
"IP address 10.10.10.1, port 3000:"

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
````diff
@@ -0,0 +1,235 @@
+```
+
+The last step to get the SensESP to report values to SignalK is to authorize it in SignalK server under security section.
+```
````
either "the MCU", "the ESP Device", or "SensESP" (without "the" before it).

tedenda replied:
Is this a better sentence?

The last step to let SensESP report measurements to Signal K server is to give it authorization.

ba58smith replied:
The last step before SensESP can send data to the server is to authorize it from the Signal K Server Admin Page.

tedenda replied:
I'll go with that.
Thanks!

ba58smith commented on examples/onewire_temperature/README.md:
````diff
@@ -0,0 +1,235 @@
+```
+
+Log in to your SignalK server and browse the menue tree:
+
````
"menu"

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+
+### Step 8 (optional)
+If you have more than one sensor connected to SensESP they will be connected to a signalk path more or less random. This means that you need to figure out what sensor is connected to what path.
+The easies way to do this is when the sensors are still on your desk, then you can simply pinch one sensor between your fingers and then look in the web GUI what sensor that has the highest temperature.
```
"Signal K", not "signalk".

"more or less at random."

Drop the first "then".

"...web GUI to see what sensor..."

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+Then mark that sensor with a label telling where it belongs. Repeat this process untill you only have one senor left.
+
+If you allready have installed all sensors in the boat and now need to get i right, you can do a similar thing but have to copy paste the adresses around until it is correct.
+
```
"already", not "allready".

If you have already installed the sensors and can't easily access them, you still have to somehow increase the temperature of a particular sensor to see which one in Signal K increases simultaneously. Then you may have to copy-paste the addresses among the sensors in the Config UI to get them all set up correctly.

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+## Verify result
+
+Last thing, verify that data is transfered to server, look in Signal-K data browser:
+
```
"Signal K", not "Signal-K".

".. by looking in the Signal K Server's "Data Browser":

tedenda replied:
Corrected

ba58smith commented on examples/onewire_temperature/README.md:
```diff
@@ -0,0 +1,235 @@
+
+## Congratulations!
+
+Well done, you have now successfully added a new SensESP to your system!
```
The device isn't called a SensESP - as noted above, it's an MCU, or an ESP Device. So the last sentence should be something like "Well done! You have successfully deployed a new ESP Device running SensESP!"

tedenda replied:
Corrected

tedenda commented:
Thank you for the review @ba58smith !

I will fix it all during the weekend.

tedenda commented:
I have now fixed most of the review items above, waiting for comments on the last ones.

I corrected several "signalk" to "Signal K" in my text but see this on other places too.
In the debug output there is a lot of "SignalK":
`␛[0m(I) (enable) Connecting SignalK source SensESP.systemhz`

Should I open a ticket for that and fix them later when I am ready with the example thing?

mairas commented:
> I have now fixed most of the review items above, waiting for comments on the last ones.
> 
> I corrected several "signalk" to "Signal K" in my text but see this on other places too.
> In the debug output there is a lot of "SignalK":
> `␛[0m(I) (enable) Connecting SignalK source SensESP.systemhz`
> 
> Should I open a ticket for that and fix them later when I am ready with the example thing?

Yea, a separate PR would be better for those.

mairas commented:
Once you're done with the modifications, could you combine the individual commits to a bit larger and more logical entities? On the command line that's done with interactive rebasing: `git rebase -i master` would do an interactive rebase for all commits after `master`. However, since you're using a Mac, I'd recommend trying out [GitUp](https://gitup.co/). I'm always having it open whenever I'm working on a Git repo. It makes it so much easier to visualize what's going on and reordering and squashing and fixupping and even splitting commits as required.

tedenda commented:
Yes I can try that, I know who I call for help if necessary 😀

tedenda commented:
> Once you're done with the modifications, could you combine the individual commits to a bit larger and more logical entities? On the command line that's done with interactive rebasing: `git rebase -i master` would do an interactive rebase for all commits after `master`. However, since you're using a Mac, I'd recommend trying out [GitUp](https://gitup.co/). I'm always having it open whenever I'm working on a Git repo. It makes it so much easier to visualize what's going on and reordering and squashing and fixupping and even splitting commits as required.

@mairas Was this what you meant?

I am happy that I made it within an hour, never done it before... Scary!

Thanks for the tip about Gitup!
In the end I used both Gitup and GitKraken to complete the task.

mairas commented:
> @mairas Was this what you meant?
> 
> I am happy that I made it within an hour, never done it before... Scary!
> 
> Thanks for the tip about Gitup!
> In the end I used both Gitup and GitKraken to complete the task.

Yes! Perfect! Thanks!

tedenda commented:
I'll fix the last bits of this PR later today.

tedenda commented:
@mairas Updated now, you can go ahead and merge when you have the time.

mairas commented:
Thanks a lot! Merging!
